src/FluffySpoon.AspNet.LetsEncrypt.Azure/AzureAppServiceSslBindingCertificatePersistenceStrategy.cs

```diff
@@ -183,24 +183,50 @@ await client.WebApps.Manager
 
 				foreach (var domain in domainsToUpgrade)
 				{
-					logger.LogDebug("Updating host name bindings for domain {0}", domain);
+					logger.LogDebug("Checking host name binding for domain {0}", domain);
 
 					if (azureOptions.Slot == null)
 					{
-						await client.WebApps.Inner.CreateOrUpdateHostNameBindingWithHttpMessagesAsync(
-							azureOptions.ResourceGroupName,
+						var existingBinding = await client.WebApps.Inner.GetHostNameBindingAsync(azureOptions.ResourceGroupName,
```

Comment:
This code is duplicated below. Can we extract it into a function somehow?

Reply:
It isn't duplicated, 2 different methods are called depending on whether a slot is in use.

Reply:
Ah nevermind, that was an oversight. Thanks!